applications/GeoMechanicsApplication/tests/cpp_tests/test_newton_raphson_strategy_linear_elastic.cpp

```diff
@@ -0,0 +1,301 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Aron Noordam
+//
+
+#include "custom_strategies/builder_and_solvers/residualbased_block_builder_and_solver_linear_elastic_dynamic.h"
+#include "custom_strategies/schemes/newmark_dynamic_U_Pw_scheme.hpp"
+#include "custom_strategies/strategies/residualbased_newton_raphson_strategy_linear_elastic_dynamic.hpp"
+#include "tests/cpp_tests/test_utilities/geo_custom_condition.h"
+#include "tests/cpp_tests/test_utilities/geo_custom_element.h"
+
+#include "geo_mechanics_fast_suite.h"
+#include "spaces/ublas_space.h"
+
+#include "factories/linear_solver_factory.h"
+#include "linear_solvers/linear_solver.h"
+
+#include "solving_strategies/convergencecriterias/displacement_criteria.h"
+
+namespace Kratos::Testing
+{
+
+using namespace Kratos;
+using SparseSpaceType  = UblasSpace<double, CompressedMatrix, Vector>;
+using LocalSpaceType   = UblasSpace<double, Matrix, Vector>;
+using LinearSolverType = LinearSolver<SparseSpaceType, LocalSpaceType>;
+
+/// <summary>
+/// Test class for the GeoMechanicNewtonRaphsonStrategyLinearElasticDynamic class
+/// The tests are based on "Finite Element Procedures, Bathe K.J., 1996, page 782"
+/// </summary>
+class NewtonRaphsonStrategyLinearElasticDynamicTester
+{
+public:
+    Model mModel;
+
+    explicit NewtonRaphsonStrategyLinearElasticDynamicTester() { CreateValidModelPart(); }
+
+    /// <summary>
+    /// Creates the valid model part with 2 degrees of freedom.
+    /// </summary>
+    void CreateValidModelPart()
```

Comment:
If the function is renamed, we don't need the comment to get the message across
```suggestion
    void CreateValidModelPartWithTwoDegreesOfFreedom()
```